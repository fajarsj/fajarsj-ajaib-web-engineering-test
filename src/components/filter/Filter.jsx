import "./Filter.scss";

const Filter = (props) => {
  const {
    handleSelectChange,
    handleSearchChange,
    handleReset,
    selectedGender,
    searchValue,
  } = props;

  return (
    <div className="filter">
      <div className="input input--group">
        <input
          type="text"
          placeholder="Search"
          className="input-control"
          value={searchValue}
          onChange={handleSearchChange}
        />
        <button className="btn btn--primary">Search</button>
      </div>
      <select
        className="input-control"
        onChange={handleSelectChange}
        value={selectedGender}
      >
        <option value="" disabled>
          Select Gender
        </option>
        <option value="all">All</option>
        <option value="male">Male</option>
        <option value="female">Female</option>
      </select>
      <button className="btn btn--outline-dark" onClick={handleReset}>
        Reset Filter
      </button>
    </div>
  );
};

export default Filter;
