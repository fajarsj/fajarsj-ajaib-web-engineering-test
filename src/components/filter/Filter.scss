.filter {
  max-width: 800px;
  display: flex;
  gap: 0.9375rem;
  margin-bottom: 1.875rem;
}

.input {
  &--group {
    display: flex;
    flex-wrap: wrap;
    align-items: stretch;
    width: 100%;

    .input-control {
      flex: 1 1 auto;
      width: auto;
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }

    .btn {
      margin-left: -1px;
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }
  }

  &-control {
    display: block;
    width: 100%;
    padding: 0.5rem 0.75rem;
    font-size: 0.875rem;
    font-weight: $normal-weight;
    color: $grey-700;
    background-color: $white;
    background-clip: padding-box;
    border: 1px solid $grey-300;
    appearance: none;
    border-radius: 0.5rem;
    transition: box-shadow 0.15s;
  }
}
