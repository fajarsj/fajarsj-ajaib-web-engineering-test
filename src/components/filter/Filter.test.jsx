import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import Filter from "./Filter";

describe("render <Filter /> correctly", () => {
  const view = render(<Filter />);

  it("renders correctly", () => {
    expect(view).toMatchSnapshot();
  });
});
