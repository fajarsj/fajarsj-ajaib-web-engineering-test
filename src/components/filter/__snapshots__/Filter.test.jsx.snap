// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Filter /> correctly renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter"
      >
        <div
          class="input input--group"
        >
          <input
            class="input-control"
            placeholder="Search"
            type="text"
            value=""
          />
          <button
            class="btn btn--primary"
          >
            Search
          </button>
        </div>
        <select
          class="input-control"
        >
          <option
            disabled=""
            value=""
          >
            Select Gender
          </option>
          <option
            value="all"
          >
            All
          </option>
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
        </select>
        <button
          class="btn btn--outline-dark"
        >
          Reset Filter
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter"
    >
      <div
        class="input input--group"
      >
        <input
          class="input-control"
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          class="btn btn--primary"
        >
          Search
        </button>
      </div>
      <select
        class="input-control"
      >
        <option
          disabled=""
          value=""
        >
          Select Gender
        </option>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="male"
        >
          Male
        </option>
        <option
          value="female"
        >
          Female
        </option>
      </select>
      <button
        class="btn btn--outline-dark"
      >
        Reset Filter
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
